### PR TITLE
chore: bump dbt to 1.7.0

### PR DESCRIPTION
**Pull request by nicor88**
# Description

Bump dbt to 1.7.0 and see what we break, close https://github.com/dbt-athena/dbt-athena/issues/454

Added functional tests for dbt docs and macro overwride.

Based on the code base we overwrite only get_catalog macro, but we don't overwrite, BaseAdapter.get_catalog(), but we overwrite a method that is called by get_catalog. 

## TODO
- [x] Test docs generation
- [x] Implement an overwride of a new macro `get_catalog_relations` that call `get_catalog_relations_where_clause_sql` (not necessary for dbt-athena, if we use glue)
   References:
   * [trino__get_catalog_relations](https://github.com/starburstdata/dbt-trino/blob/2e0cd5854f14127d59ac4fb356a67dde7884fce0/dbt/include/trino/macros/catalog.sql#L23C10-L42)
   * [trino__get_catalog_relations_where_clause_sql](https://github.com/starburstdata/dbt-trino/blob/2e0cd5854f14127d59ac4fb356a67dde7884fce0/dbt/include/trino/macros/catalog.sql#L151-L175) - this is not really necessary if we use glue APIs

## Checklist

- [x] You followed [contributing section](https://github.com/dbt-athena/dbt-athena#contributing)
- [x] You kept your Pull Request small and focused on a single feature or bug fix.
- [x] You added unit testing when necessary
- [x] You added functional testing when necessary

### CODE REVIEW

File: .github/workflows/functional-tests-workflow.yml [hunk 1]
```diff
@@ -35,10 +35,10 @@ jobs:
         with:
           ref: ${{ inputs.checkout-ref }}
           repository: ${{ inputs.checkout-repository }}
-      - name: Set up Python 3.10
+      - name: Set up Python 3.11
```
Comment (svdimchenko) [inline .github/workflows/functional-tests-workflow.yml:38]: BTW why do we use 3.11 for functional tests instead of 3.8 ? I believe using 3.8 may help us to avoid issues like this https://github.com/dbt-athena/dbt-athena/issues/483. Did I miss smth ?

Comment (nicor88) [reply .github/workflows/functional-tests-workflow.yml:38]: 3.11 is slightly faster, so I put 3.11 only for that. 
But I'm happy to switch to 3.8, to cover edge cases.

**Review by svdimchenko — APPROVED** [review]

File: dbt/adapters/athena/impl.py [hunk 2]
```diff
@@ -658,6 +665,26 @@ def list_relations_without_caching(self, schema_relation: AthenaRelation) -> Lis
 
         return relations
 
+    def _get_one_catalog_by_relations(
```
[the remark this answers is not in the document]
Comment (nicor88) [reply dbt/adapters/athena/impl.py:668]: TODO: add unit test for this function.

File: dbt/adapters/athena/impl.py [hunk 3]
```diff
@@ -658,6 +665,26 @@ def list_relations_without_caching(self, schema_relation: AthenaRelation) -> Lis
...
+        self,
+        information_schema: InformationSchema,
+        relations: List[BaseRelation],
+        manifest: Manifest,
+    ) -> agate.Table:
+        """
+        Overwrite of _get_one_catalog_by_relations for Athena, in order to use glue apis.
+        This function is invoked by Adapter.get_catalog_by_relations. And
```
Comment (svdimchenko) [inline dbt/adapters/athena/impl.py:676]: seems the docstring should be finalized

File: dbt/adapters/athena/impl.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (svdimchenko) [reply dbt/adapters/athena/impl.py:668]: can we add unit tests now ? seems we can cover it with moto

[the remark this answers is not in the document]
Comment (nicor88) [reply dbt/adapters/athena/impl.py:668]: yes, that's why I left a TODO

File: dbt/adapters/athena/impl.py [hunk 4]
```diff
@@ -658,6 +665,28 @@ def list_relations_without_caching(self, schema_relation: AthenaRelation) -> Lis
...
+        _table_definitions = []
+        for _rel in relations:
+            glue_table_definition = self.get_glue_table(_rel)
+            if glue_table_definition:
+                _table_definition = self._get_one_table_for_catalog(glue_table_definition["Table"], _rel.database)
+                print(_table_definition)
+                _table_definitions.append(_table_definition)
+        table = agate.Table.from_object(_table_definitions)
+        # TODO this doesn't work, must be fixed
+        # return self._join_catalog_table_owners(table, manifest)
```
Comment (nicor88) [inline dbt/adapters/athena/impl.py:687]: @Jrmyy table contains the output of _get_one_table_for_catalog - actually one entry per column.

I want to use _join_catalog_table_owners to enrich the table with the owner, but somehow I get: 
`E       KeyError: 'table_database'`

_table_definitions is:
```
[{'table_database': 'awsdatacatalog', 'table_schema': 'foo', 'table_name': 'bar', 'table_type': 'table', 'table_comment': '', 'column_name': 'id', 'column_index': 0, 'column_type': 'string', 'column_comment': ''}, {'table_database': 'awsdatacatalog', 'table_schema': 'foo', 'table_name': 'bar', 'table_type': 'table', 'table_comment': '', 'column_name': 'country', 'column_index': 1, 'column_type': 'string', 'column_comment': ''}, {'table_database': 'awsdatacatalog', 'table_schema': 'foo', 'table_name': 'bar', 'table_type': 'table', 'table_comment': '', 'column_name': 'dt', 'column_index': 2, 'column_type': 'date', 'column_comment': ''}]
```

File: dbt/adapters/athena/impl.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (nicor88) [reply dbt/adapters/athena/impl.py:668]: @svdimchenko I added some unit tests for this function, similar to what we do for `test__get_one_catalog`, and they are really similar, except that _get_one_catalog_by_relations receive a list of relations to filter out table metadata.

File: dbt/adapters/athena/impl.py [hunk 5]
```diff
@@ -455,6 +456,8 @@ def _s3_path_exists(self, s3_bucket: str, s3_prefix: str) -> bool:
         return True if "Contents" in response else False
 
     def _join_catalog_table_owners(self, table: agate.Table, manifest: Manifest) -> agate.Table:
+        # print(manifest)
+        # print(table)
```
Comment (svdimchenko) [inline dbt/adapters/athena/impl.py:460]:
```suggestion
```

File: dbt/adapters/athena/impl.py [hunk 6]
```diff
@@ -469,6 +472,9 @@ def _join_catalog_table_owners(self, table: agate.Table, manifest: Manifest) ->
                 )
         owners_table = agate.Table.from_object(owners)
 
+        print(table)
+        print(owners_table)
+
```
Comment (svdimchenko) [inline dbt/adapters/athena/impl.py:477]:
```suggestion
```

File: dbt/adapters/athena/impl.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (svdimchenko) [reply dbt/adapters/athena/impl.py:674]: @nicor88 great job done! 💪

**Review by svdimchenko — COMMENTED** [review]
@nicor88 it's great you've added dbt-core 1.7.0 support! I've left just minor comments about commented out code

**Comment by nicor88** [comment]
@Jrmyy and @svdimchenko another review please 🙏🏻

**Review by svdimchenko — APPROVED** [review]

File: .github/workflows/functional-tests-workflow.yml [hunk 7]
```diff
@@ -35,10 +35,10 @@ jobs:
         with:
           ref: ${{ inputs.checkout-ref }}
           repository: ${{ inputs.checkout-repository }}
-      - name: Set up Python 3.10
+      - name: Set up Python
         uses: actions/setup-python@v4
         with:
-          python-version: '3.10'
+          python-version: '3.8'
```
Comment (Jrmyy) [inline .github/workflows/functional-tests-workflow.yml:41]: Why downgrading to 3.8 ?

**Review by Jrmyy — APPROVED** [review]
Nice to add tests ✅

File: .github/workflows/functional-tests-workflow.yml (hunk 7 above)
Comment (nicor88) [reply .github/workflows/functional-tests-workflow.yml:41]: Originally I put 3.11 because it was slightly faster, but @svdimchenko suggested the oldest version to be sure that all works as expected. Generally older versions are the most problematic, as we tend to use the newest python capabilities.
